### ISSUE TITLE
Change to protected_environment deploy_access_levels is not detected

### ISSUE THREAD

**Issue by hertzsprung** [gitlabform/gitlabform#644]
**Describe the bug**

Changing gitlabform protected_environment deploy_access_levels from `user` to `access_level` does nothing (debug logs erroneously state: "myenv of protected_environments in mygroup/myproject doesn't need an update"

Config before:
```
mygroup/myproject:
  protected_environments:
    enforce: true
    myenv:
      deploy_access_levels:
        - user: my-user
```

Config after:
```
mygroup/myproject:
  protected_environments:
    enforce: true
    myenv:
      deploy_access_levels:
        - access_level: developer
```

**GitLabForm version**

3.8.0

**GitLab version**

GitLab Enterprise Edition 16.7.0-pre [65594f41831](https://gitlab.com/gitlab-org/gitlab/-/commits/65594f41831)

**Comment by amimas** [comment]
I assune the gitlab version 16.7 is a typo. As of right now, I think latest version is 16.6. If this is not a typo, is it possible you're working with an unreleased gitlab version that has a regression bug?

**Comment by amimas** [comment]
@hertzsprung - any update available from you on this?

I was able to reproduce this in PR #857 when added new acceptance tests. Please see the description of that PR. It seems to be a **GitLab Issue**. The current work-around I can see is to simply re-run gitlabform again a 2nd time. Or, it might be addressed if the above PR is merged and released with a performance hit.

**Comment by amimas** [comment]
@hertzsprung - The PR mentioned above has been merged and released in v4.0.1. It should be resolved now. Closing this as fixed.
